sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -712,11 +719,11 @@ abstract class BuildableElement<T extends Widget> extends Element<T> {
   /// components dirty during event handlers before the frame begins, not during
   /// the build itself.
   void markNeedsBuild() {
-    assert(!_debugStateLocked);
     assert(_debugLifecycleState == _ElementLifecycle.mounted);
     if (_dirty)
       return;
     _dirty = true;
+    assert(!_debugInBuildScope);
```

Comment:
I'm playing the same trick for didUpdateConfig, which means tracking the currently building element isn't sufficient.  It looks like we're doing something similar for initState.  I've instead made them all work the same way.